File: Round 2 2008/B.py
#!/usr/bin/python
#
# Problem: Triangle Areas
# Language: Python
# Author: KirarinSnow
# Usage: python thisfile.py <input.in >output.out
# Comments: Inefficient brute-force solution. 


from math import *

def compute():
    n, m, a = map(int, raw_input().split())

    xa = 0
    ya = 0
    for xb in range(0, n+1):
	if xb == 0:
	    v = 0
	else:
	    v = a/xb
	for yc in range(v, m+1):
	    bc = xb * yc - a
	    xc = 0
	    yb = 0
	    if bc == 0:
		xc = 0
		yb = 0
	    else:
		for k in range(1, int(sqrt(max(0,bc)))+1): 
		    if bc % k == 0 and bc/k <= n and k <= m:
			xc = bc/k
			yb = k
			break

	    if xb*yc-xc*yb == a:
                return ' '.join(map(str, [xa, ya, xb, yb, xc, yc]))


    return "IMPOSSIBLE"

for i in range(input()):
    print "Case #%d: %s" % (i+1, compute())

File: Round 2 2008/B.pl
#!/usr/bin/perl
#
# Problem: Triangle Areas
# Language: Perl
# Author: KirarinSnow
# Usage: perl thisfile.pl <input.in >output.out


sub compute()
{
    ($n, $m, $a) = split(/ /, <>);
       
    if ($a > $n * $m)
    {
	return "IMPOSSIBLE";
    }
    else
    {
	$x = ($a - 1)%$n + 1;
	$y = ($a - 1)/$n + 1;
	return sprintf("0 1 $n 0 $x %d", $y);
    }
}

$cases = <>;

for ($i = 1; $i <= $cases; $i++)
{
    $o = compute();
    print "Case #$i: $o\n";
}
